planner/languages/nginx/nginx_planner.go

```diff
@@ -45,11 +43,12 @@ func (p *Planner) GetPlan(srcDir string) *plansdk.Plan {
 			Command:    fmt.Sprintf(startCommand, p.buildConfig(srcDir)),
 			InputFiles: plansdk.AllFiles(),
 		},
-		Definitions: []string{
-			fmt.Sprintf(nginxShellStartScript, srcDir, p.shellConfig(srcDir)),
-		},
 		GeneratedFiles: map[string]string{
 			"shell-helper-nginx.conf": fmt.Sprintf(shellHelperNginxConfig, os.TempDir()),
+			"shell-nginx.sh":          fmt.Sprintf(nginxShellStartScript, srcDir, p.shellConfig(srcDir)),
+		},
+		ShellInitHook: []string{
+			fmt.Sprintf(". %s", filepath.Join(srcDir, ".devbox/gen/shell-nginx.sh")),
```

Comment:
I'm a bit unsure about splitting this out in this way. It feels a bit fragile. I almost want an option that does both (create a script and either calls it or sources it in the init hook)

Reply:
Also, this should probably not be called `shell-nginx.sh` it it's an init script. Maybe just `init.sh` ?